### PR TITLE
Add 2 modules to create users in auth0 domain and to gather facts abo…

### PR DESCRIPTION
**Pull request by haad**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

auth0_user
auth0_facts
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

This PR is adding two new modules which can be used to manage users on auth0. Create, remove, update their metadata/pass etc.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @haad for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: web_infrastructure/auth0_user.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      result = json.loads(r.text)
+    else:
+      result = "Rest call has not returned any data."
+
+    module.exit_json(results=result)
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py:164]: Please add the following and remove the `import requests`
`from ansible.module_utils.urls import *`

For more information please search for `ansible.module_utils.urls` on http://docs.ansible.com/ansible/developing_modules.html#module-checklist

File: web_infrastructure/auth0_user_facts.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    if r.status_code != 200:
+      module.fail_json(msg='Request to Auth0 failed with return code %s, reason: %s'%(r.status_code, r.reason))
+
+    module.exit_json(results=json.loads(r.text))
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
```
Comment (gundalow) [inline web_infrastructure/auth0_user_facts.py:106]: Please add the following and remove the `import requests`
`from ansible.module_utils.urls import *`

For more information please search for `ansible.module_utils.urls` on http://docs.ansible.com/ansible/developing_modules.html#module-checklist

File: web_infrastructure/auth0_user.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+DOCUMENTATION = '''
+---
+module: auth0_user
+short_description: Module for managing auth0 users
+description:
+    - Auth0 user managing module. It can search for a given user and
+      return it's payload, create/delete user and update application
+      metadata.
+requirements: [ python requests ]
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: The requirements line can be removed as Ansible will automatically install `requests` if needed, when using `ansible.module_utils.urls`

File: web_infrastructure/auth0_user_facts.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+DOCUMENTATION = '''
+---
+module: auth0_user_facts
+short_description: Module for managing auth0 users
+description:
+    - Auth0 user managing module. It can search for a given user and
+      return it's payload, create/delete user and update application
+      metadata.
+requirements: [ python requests ]
```
Comment (gundalow) [inline web_infrastructure/auth0_user_facts.py]: The requirements line can be removed as Ansible will automatically install `requests` if needed, when using `ansible.module_utils.urls`

File: web_infrastructure/auth0_user.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+#only defaults
+'''
+
+import sys
+import requests
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth0_token=dict(required=True, type='str', aliases=['token']),
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: For password / secret arguments `no_log=True` should be set.

File: web_infrastructure/auth0_user_facts.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+#only defaults
+'''
+
+import sys
+import requests
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth0_token=dict(required=True, type='str'),
```
Comment (gundalow) [inline web_infrastructure/auth0_user_facts.py]: For password / secret arguments `no_log=True` should be set.

**Comment by gundalow** [comment]
All descriptions should be sentences (capital letter & full stop)

File: web_infrastructure/auth0_user.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+description:
+    - Auth0 user managing module. It can search for a given user and
+      return it's payload, create/delete user and update application
+      metadata.
+requirements: [ python requests ]
+version_added: "2.2"
+options:
+  state:
+    description:
+     - the state of the collection. Either present or absent
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: 1) You don't need the `Either present or absent`, the `choices` section will be automatically added to the documentation

2) Is it worth defining how 'delete_all' is different to 'absent'

File: web_infrastructure/auth0_user.py [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+import sys
+import requests
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth0_token=dict(required=True, type='str', aliases=['token']),
+            auth0_domain=dict(required=True, type='str', aliases=['domain']),
+            state=dict(choices=['present', 'absent', 'delete_all']),
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: Missing `required=True`

File: web_infrastructure/auth0_user.py [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+      metadata.
+requirements: [ python requests ]
+version_added: "2.2"
+options:
+  state:
+    description:
+     - the state of the collection. Either present or absent
+    default: 'present'
+    required: true
+    aliases: []
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: You can remove the empty `aliases` section from both modules

File: web_infrastructure/auth0_user.py [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    choices: ['present', 'absent', 'delete_all']
+  user_data:
+    description:
+     - data used to update given user definition
+    required: false
+    aliases: ['body']
+  user_id:
+    description:
+     - data used to update given user definition
```
Comment (gundalow) [inline web_infrastructure/auth0_user.py]: Copy and past of user_data?

File: web_infrastructure/auth0_user_facts.py [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    - Auth0 user managing module. It can search for a given user and
+      return it's payload, create/delete user and update application
+      metadata.
+requirements: [ python requests ]
+version_added: "2.2"
+options:
+  query:
+    description:
+     - query to select given user
+    required: false
```
Comment (gundalow) [inline web_infrastructure/auth0_user_facts.py]: Later on you've defined this as required

**Comment by gundalow** [comment]
needs_revision

**Comment by haad** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @haad for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @haad for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@haad A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@haad Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by haad** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @haad for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**